src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -544,8 +591,178 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
+                        auto contour = std::begin(pcs.upper);
+                        const auto c1_area = std::abs(contour->get().Get_Signed_Area());
+                        ++contour;
+                        const auto c2_area = std::abs(contour->get().Get_Signed_Area());
+
+                        //cap the smaller contour and tile the larger contoour with the lower plane contour
+                        if  (c1_area < c2_area){
+                            close_hole_in_floor(pcs.upper.front());
+
+                            auto new_faces = Estimate_Contour_Correspondence(pcs.upper.back(), pcs.lower.front());
+
+                            const auto old_face_count = amesh.vertices.size();
+                            for(const auto &p : pcs.upper.back().get().points) amesh.vertices.emplace_back(p);
+                            for(const auto &p : pcs.lower.front().get().points) amesh.vertices.emplace_back(p);
+                            for(const auto &fs : new_faces){
+                               const auto f_A = static_cast<uint64_t>(fs[0] + old_face_count);
+                               const auto f_B = static_cast<uint64_t>(fs[1] + old_face_count);
+                               const auto f_C = static_cast<uint64_t>(fs[2] + old_face_count);
+                               amesh.faces.emplace_back( std::vector<uint64_t>{{f_A, f_B, f_C}} );
+                            }
+
+                        }else{
+                            close_hole_in_floor(pcs.upper.back());
+
+                            auto new_faces = Estimate_Contour_Correspondence(pcs.upper.front(), pcs.lower.front());
+
+                            const auto old_face_count = amesh.vertices.size();
+                            for(const auto &p : pcs.upper.front().get().points) amesh.vertices.emplace_back(p);
+                            for(const auto &p : pcs.lower.front().get().points) amesh.vertices.emplace_back(p);
+                            for(const auto &fs : new_faces){
+                               const auto f_A = static_cast<uint64_t>(fs[0] + old_face_count);
+                               const auto f_B = static_cast<uint64_t>(fs[1] + old_face_count);
+                               const auto f_C = static_cast<uint64_t>(fs[2] + old_face_count);
+                               amesh.faces.emplace_back( std::vector<uint64_t>{{f_A, f_B, f_C}} );
+                            }
+                        }
+
+                    }else{
+                        //do two-to-one branching
+                    }
+
+                }else if( (N_upper == 1) && (N_lower == 2) ){
+                    //check if the lower plane contains enclosed contours
+                    if (projected_contours_overlap(*l_cp_it, pcs.lower.front(),*m_cp_it,pcs.lower.back())
+                        && !projected_contours_intersect(*l_cp_it, pcs.lower.front(),*m_cp_it,pcs.lower.back())){
```

Comment:
this seems like a copy of the previous code, only with upper and lower switched. Can we create a lambda function? that way it would work with mine as well